### ISSUE TITLE
change shebang to be more compatible

### ISSUE THREAD

**Issue by wucke13** [serde-rs/serde#2583]
This

https://github.com/serde-rs/serde/blob/0676673ca5efef667d10a7b63b5b078a8d64e43a/precompiled/build.sh#L1

should be

```bash
#!/usr/bin/env bash
````

in order to maximize compatibility. On many more nieche systems `/bin/bash` does not exist (`/bin/sh` would be fine!), but `/usr/bin/env` and `env` would be able to find `bash` on most if not all systems _if_ `bash` is present at all.

**Comment by oli-obk** [comment]
Good catch. I'd merge a PR that changes this

**Comment by wucke13** [comment]
No problem. Another friction point with the script is that it only works with a rustup cargo, not with native cargo (due to reliance on `+nightly`). I don't know how to robustly detect if `cargo` is infact `rustup`:

```console
+ cargo +nightly build --manifest-path bin/Cargo.toml --bin serde_derive --profile precompiled -Z unstable-options -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort --target x86_64-unknown-linux-musl --out-dir serde_derive
error: no such command: `+nightly`

	Cargo does not handle `+toolchain` directives.
	Did you mean to invoke `cargo` through `rustup` instead?
````

**Comment by oli-obk** [comment]
I think the rustup issue will be fixed by whatever resolves #2575
